repose-aggregator/src/docs/asciidoc/recipes/rate-limiting.adoc

```diff
@@ -1,72 +1,124 @@
-    </destinations>
-  </repose-cluster>
+    ...
 </system-model>
 ----
 
-In the following example, the Repose instance with the `node` `id` of `node1` will launch a Distributed Datastore service which will listen on port 9999.
-The Repose instance with the `node` `id` of `node2` will launch it's own Distributed Datastore service which will listen on port 7777.
-With `allow-all` set to `false`, a host that is not in the cluster cannot communicate with the Distributed Datastore.
+=== Rate Limiting
+After the filter has been added to the <<../architecture/system-model.adoc#,System Model>>, the example configuration can copied/moved from the examples directory to the configuration files directory (i.e. `/etc/repose`).
+The example configuration for rate limiting is sufficient for testing and will limit requests to the origin service to 10 times per minute.
 
 [source,xml]
-.dist-datastore.cfg.xml
+.rate-limiting.cfg.xml
 ----
-<?xml version="1.0" encoding="UTF-8"?>
-<distributed-datastore xmlns='http://docs.openrepose.org/repose/distributed-datastore/v1.0'>
-    <allowed-hosts allow-all="false"/>
-    <port-config>
-        <port port="9999" cluster="repose"/>
-        <port port="7777" cluster="repose" node="node2"/>
-    </port-config>
-</distributed-datastore>
+include::https://raw.githubusercontent.com/rackerlabs/repose/master/repose-aggregator/components/filters/rate-limiting-filter/src/main/resources/META-INF/schema/examples/rate-limiting.cfg.xml[]
 ----
 
-[source,xml]
-.rate-limiting.cfg.xml
+== Testing
+=== Script
+You can use this script to quickly make 11 requests to *Repose* to confirm that rate limiting is working.
+
+[source,shell]
+.test_rate_limiting.sh
+----
+#!/bin/bash
+OUT_FILE="repose-curl.out"
+rm -f $OUT_FILE
+touch $OUT_FILE
+for i in {1..11} ; do
+  echo -en "\n\n~~~~~ Attempt #$i ~~~~~\n\n" >> $OUT_FILE
+  curl -H "x-pp-user: abc123" -H "Content-Type: Test" -H "Content-Length: 0" localhost:8080/get -v >> $OUT_FILE 2>&1
+done
 ----
-<?xml version="1.0" encoding="UTF-8"?>
-<rate-limiting xmlns="http://docs.openrepose.org/repose/rate-limiting/v1.0" datastore="distributed/hash-ring">
-    <!-- Protects the Origin Service from being flooded. -->
-    <global-limit-group>
-        <limit id="global" uri="*" uri-regex=".*" value="1000" unit="MINUTE"/>
-    </global-limit-group>
 
-    <!-- Limits for all other requests -->
-    <limit-group id="limited" groups="IP_Standard" default="true">
-        <limit id="all" uri="*" uri-regex=".*" http-methods="POST PUT GET DELETE" value="10" unit="MINUTE"/>
-    </limit-group>
+=== Running
+Assuming you named the script `test_rate_limiting.sh`, you can run it with the following:
 
-    <!-- Limits for WhiteListed IPs -->
-    <limit-group id="unlimited" groups="IP_Super" default="false"/>
-</rate-limiting>
+[source,shell]
+.console session
 ----
+# make the script executable
+chmod +x test_rate_limiting.sh
+
+# run the script
+./test_rate_limiting.sh
```

Comment:
Since you are assuming the working directory is the directory containing the `test_rate_limiting.sh` anyway (from the previous command), why not drop the leading `./`?

Reply:
Unless they have the current working directory in their path (which is bad by the way), then you must include the `./` to make it go.

Reply:
Oh derp. That's right. The shell resolves the pathing bits to give you an absolute path, doesn't it? So the script is not on the `PATH`, and thus, must be run from an absolute path, and the `./` is a shorthand for getting that. Not sure what I was thinking.

Reply:
So I guess the alternative would be:
`sh test_rate_limiting.sh`

... but that's not really any better. So I guess I don't care.